src/rpc/client/accessibility.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ * Modifications copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PageChannel } from '../channels';
+import { ElementHandle } from './elementHandle';
+import * as types from '../../types';
+
+export class Accessibility {
+  private _channel: PageChannel;
+
+  constructor(channel: PageChannel) {
+    this._channel = channel;
+  }
+
+  snapshot(options: { interestingOnly?: boolean; root?: ElementHandle } = {}): Promise<types.SerializedAXNode | null> {
+    const root = options.root ? options.root._elementChannel : undefined;
+    return this._channel.accessibilitySnapshot({ options: { interestingOnly: options.interestingOnly, root } });
```

Comment:
Shouldn't the recursive handle->guid unwrapper handle the `root`?

Reply:
I need to go from ElementHandle to ElementHandleChannel here, recursive wrapper goes Channel -> guid